sdk/iot/azure-iot-modelsrepository/azure/iot/modelsrepository/client.py

```diff
@@ -0,0 +1,173 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import six.moves.urllib as urllib
+import re
+from azure.core import PipelineClient
+from azure.core.pipeline.transport import RequestsTransport
+from azure.core.configuration import Configuration
+from azure.core.pipeline.policies import (
+    UserAgentPolicy,
+    HeadersPolicy,
+    RetryPolicy,
+    RedirectPolicy,
+    BearerTokenCredentialPolicy,
+    ContentDecodePolicy,
+    NetworkTraceLoggingPolicy,
+    ProxyPolicy,
+)
+from . import resolver as resolver
+from . import pseudo_parser
+from .chainable_exception import ChainableException
+
+
+# Public constants exposed to consumers
+DEPENDENCY_MODE_TRY_FROM_EXPANDED = "tryFromExpanded"
+DEPENDENCY_MODE_DISABLED = "disabled"
+DEPENDENCY_MODE_ENABLED = "enabled"
+
+
+# Convention-private constants
+_DEFAULT_LOCATION = "https://devicemodels.azure.com"
+_DEFAULT_API_VERSION = "2021-02-11"
+_REMOTE_PROTOCOLS = ["http", "https"]
+
+
+class ModelsRepositoryClient(object):
+    """Client providing APIs for Models Repository operations"""
+
+    def __init__(self, repository_location=None, api_version=None, **kwargs):
+        """
+        :param str repository_location: Location of the Models Repository you wish to access.
+            This location can be a remote HTTP/HTTPS URL, or a local filesystem path.
+            If omitted, will default to using "https://devicemodels.azure.com".
+
+        :raises: ValueError if repository_location is invalid
+        """
+        repository_location = (
+            _DEFAULT_LOCATION if repository_location is None else repository_location
+        )
+        # api_version = _DEFAULT_API_VERSION if api_version is None else api_version
+
+        kwargs.setdefault("api_verison", api_version)
+
+        # NOTE: depending on how this class develops over time, may need to adjust relationship
+        # between some of these objects
+        self.fetcher = _create_fetcher(
+            location=repository_location, api_version=api_version, **kwargs
+        )
+        self.resolver = resolver.DtmiResolver(self.fetcher)
+        self.pseudo_parser = pseudo_parser.PseudoParser(self.resolver)
+
+    def get_models(self, dtmis, dependency_resolution=DEPENDENCY_MODE_DISABLED):
```

Comment:
Would be good to support [distributed tracing](https://azure.github.io/azure-sdk/python_implementation.html#distributed-tracing) for client service methods (which only includes `get_models` in this first iteration goal) and apply the patterns where we need to.